source/extensions/filters/network/redis_proxy/command_splitter_impl.cc

```diff
@@ -369,7 +374,7 @@ SplitRequestPtr mgmtNoKeyRequest::create(Router& router, Common::Redis::RespValu
     if(shard_index < 0){
       shard_index = i;
     }
-    request_ptr->pending_requests_.emplace_back(*request_ptr, i,shard_index,getresponseHandlerType(command_name));
+    request_ptr->pending_requests_.emplace_back(*request_ptr, i, shard_index, command_name, firstarg, getresponseHandlerType(command_name));
```

Comment:
Instead of passing arguments , custom handler types can be introduced for each subcommand category

Reply:
Will raise ticket to address this ji